### PR TITLE
consider view for legend representatives

### LINKED ISSUE #10935: Dask compat issues w/ nullable
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)

bokeh 2.3.0rc2, dask 2021.2.0

#### Stack traceback and/or browser JavaScript console output

Screenshot 2021-02-23 at 1 56 30 AM

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #10935
- [x] tests added / passed

@mattpap this fails the test with the expected out-of-bounds error without the changes, and passes the test with the changes. It also behaves well with more extensive dask operations e.g. 
```
In [2]: import dask.array as da
   ...:
   ...: X = da.random.random((2000000, 100), chunks=(100000, 100)).persist()
   ...: u, s, v = da.linalg.svd_compressed(X, k=5)

In [3]: v.compute()
```
and as far as I can tell, the graph looks correct (nodes later after "processing" nodes are marked "waiting" etc.) cc @quasiben 

I am sure the code could be cleaner or make use of better APIs I am just unfamiliar with. Please do one of two things first things in the morning:

* make any small changes or improvements, then merge, or
* merge as-is if you are content with it

I do still plan to release tomorrow with this fix included.

### CODE REVIEW

**Comment by bryevdv** [comment]
We might want to look over old issues in case there are any legend/view issues that this or your PR also fix.

File: bokehjs/test/unit/regressions.ts [hunk 1]
```diff
@@ -91,4 +91,17 @@ describe("Bug", () => {
       await display(plot)
     })
   })
+
+  describe("in issue #10935 (2)", () => {
```
Comment (mattpap) [inline bokehjs/test/unit/regressions.ts:95]: Two bugs one issue, then nest under one `describe` branch.

File: bokehjs/test/unit/regressions.ts [hunk 2]
```diff
@@ -91,4 +91,17 @@ describe("Bug", () => {
...
+    it("prevents to render a legend and an empty view", async () => {
```
Comment (mattpap) [inline bokehjs/test/unit/regressions.ts:96]: Is this an accurate description? I think it should be a subset of indices.

File: bokehjs/src/lib/models/renderers/glyph_renderer.ts [hunk 3]
```diff
@@ -423,9 +423,18 @@ export class GlyphRenderer extends DataRenderer {
     if (field != null) {
       const data = this.data_source.get_column(field)
       if (data != null) {
-        const i = indexOf(data, value)
-        if (i != -1)
-          index = i
+        if (this.view == null) {
```
Comment (mattpap) [inline bokehjs/src/lib/models/renderers/glyph_renderer.ts:426]: `view` is non-nullable, so this should not happen.

**Comment by bryevdv** [comment]
@mattpap can you can make the small changes and then merge? I was hoping to start cutting the release immediately after the meeting.

**Pull request merged**